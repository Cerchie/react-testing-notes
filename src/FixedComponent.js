import React from 'react';

function FixedComponent(props){
    return (
        <>
        <p>{props.favNum}</p>
        <p>{props.favNum + 2}</p>
        <p>{props.favNum + 1}</p>
        </>
    )
}

FixedComponent.defaultProps = {
    favNum: 42
}

export default FixedComponent;