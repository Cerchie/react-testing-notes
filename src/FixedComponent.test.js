import React from 'react';
import { render, screen } from '@testing-library/react';
import FixedComponent from './FixedComponent'
it('renders w/o crashing', ()=> {
  render(<FixedComponent />);
})

it('matches snapshot', function(){
    const {asFragment} = render(<FixedComponent/>);
    expect(asFragment()).toMatchSnapshot()
});