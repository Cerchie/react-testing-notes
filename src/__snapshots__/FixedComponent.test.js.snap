// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p>
    42
  </p>
  <p>
    44
  </p>
  <p>
    43
  </p>
</DocumentFragment>
`;
